src/librustc_mir/build/matches/mod.rs

```diff
@@ -361,15 +358,17 @@ impl<'a,'tcx> Builder<'a,'tcx> {
 
         // If all candidates were sorted into `target_candidates` somewhere, then
         // the initial set was inexhaustive.
-        let untested_candidates = candidates.len() - tested_candidates;
+        let untested_candidates = unmatched_candidates.len() - tested_candidates;
```

Comment:
Maybe do a `let untestd_candidates = unmatched_candidates.split_off(tested_candidates)` here and then next check becomes `if untested_candidates.len() == 0`.